eng/SourceBuild.Tarball.targets

```diff
@@ -0,0 +1,375 @@
+        Include="$(DotNetCliToolDir)**"
+        Exclude="
+          $(DotNetCliToolDir)shared\2.1.0\**;
+          $(DotNetCliToolDir)tools\**;
+          "
+        Condition="'$(CustomDotNetSdkDir)' == ''"/>
+
+      <TarballCopyFile
+        Include="@(TarballRootToolFile)"
+        RelativeDestination="$([MSBuild]::MakeRelative('$(ProjectDir)', '%(Identity)'))" />
+
+      <TarballGitInfoFile Include="$(ProjectDir)artifacts\git-info\**" />
+      <TarballCopyFile
+        Include="@(TarballGitInfoFile)"
+        RelativeDestination="git-info\%(RecursiveDir)%(Filename)%(Extension)" />
+
+      <TarballBootstrappingToolFile Include="$(ProjectDir)support\tarball\tool-bootstrapping/**;" />
+      <TarballCopyFile
+        Include="@(TarballBootstrappingToolFile)"
+        RelativeDestination="tool-bootstrapping\%(RecursiveDir)%(Filename)%(Extension)" />
+
+      <TarballCopyFile
+        Include="$(ProjectDir)support/tarball/build.sh"
+        RelativeDestination="build.sh" />
+
+      <!-- Setup package version props to include both source-built and running PackageVersions.props -->
+      <TarballCopyFile
+        Include="$(ProjectDir)support\tarball\PackageVersions.props"
+        RelativeRoot="artifacts/obj/$(Platform)/Release/" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="AddTarballPackages">
+    <!-- Remove reference-packages from tarball prebuilts...-->
+    <ItemGroup Condition="'$(CustomRefPackagesDir)' != ''">
+      <ExcludePrebuiltPackageFile Include="$(CustomRefPackagesDir)**\*.nupkg" />
+    </ItemGroup>
+
+    <Exec
+      Command="tar -tf $(ExternalTarballsDir)Private.SourceBuild.ReferencePackages.*.tar.gz | tr '[:upper:]' '[:lower:]'"
+      ConsoleToMsBuild="true"
+      StandardOutputImportance="low"
+      Condition="'$(CustomRefPackagesDir)' == ''">
+      <Output TaskParameter="ConsoleOutput" ItemName="SourceBuildReferencePackageFile" />
+    </Exec>
+
+    <!-- Remove previously source-built packages from tarball prebuilts... -->
+    <ItemGroup Condition="'$(CustomPreviouslySourceBuiltPackagesDir)' != ''">
+      <ExcludePrebuiltPackageFile Include="$(CustomPreviouslySourceBuiltPackagesDir)**\*.nupkg" />
+    </ItemGroup>
+
+    <Exec
+      Command="tar -tf $(ExternalTarballsDir)Private.SourceBuilt.Artifacts.*.tar.gz | tr '[:upper:]' '[:lower:]'"
+      ConsoleToMsBuild="true"
+      StandardOutputImportance="low"
+      Condition="'$(CustomPreviouslySourceBuiltPackagesDir)' == ''">
+      <Output TaskParameter="ConsoleOutput" ItemName="PreviouslySourceBuiltPackageFile" />
+    </Exec>
+
+    <!-- Add packages to the tarball. -->
+    <ItemGroup>
+      <!-- Set up prebuilt nupkg filenames from tarballs to exclude. -->
+      <SourceBuildReferencePackageFilename Include="@(SourceBuildReferencePackageFile -> '%(Filename)')" />
+      <PreviouslySourceBuiltPackageFilename Include="@(PreviouslySourceBuiltPackageFile -> '%(Filename)')" />
+
+      <!-- Remove source-built packages from tarball prebuilts... -->
```

Comment:
This comment confused me.  It's not really removing these packages from the tarball prebuilts, it's just collecting the names so it can remove them later.  I think the remove comment should be later when they're removed.

Reply:
These are just copied from the tarball script. Reworked to "*Find* ... packages *to exclude* from tarball prebuilts", since we don't really remove them anymore.